### PR TITLE
8257638: Update usage of "type" terminology in javax.lang.model

### PR DESCRIPTION
**Pull request by jddarcy**
First pass at updating javax.lang.model's usage of the word "type" to correspond to changes in usage of "type" being made in JLS.

The wording in annotated construct and related functionality may need another pass.

Updating @jls tags to renamed JLS sections will be done under a different bug id.

Paragraphs *not* reflowed initially to easy reviewing. Paragraph will be reflowed and copyright year updated before a push is done.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8257638](https://bugs.openjdk.java.net/browse/JDK-8257638): Update usage of "type" terminology in javax.lang.model

### Reviewers
 * [Jonathan Gibbons](https://openjdk.java.net/census#jjg) (@jonathan-gibbons - **Reviewer**) ⚠️ Review applies to 66e9f45ff31186a532e5f5a43a1a2a1737eb72dd

### Download
`$ git fetch https://git.openjdk.java.net/jdk pull/1726/head:pull/1726`
`$ git checkout pull/1726`

### CODE REVIEW

File: src/java.compiler/share/classes/javax/annotation/processing/Filer.java [hunk 1]
```diff
@@ -66,7 +66,7 @@
  * <p>The file creation methods take a variable number of arguments to
  * allow the <em>originating elements</em> to be provided as hints to
  * the tool infrastructure to better manage dependencies.  The
- * originating elements are the types or packages (representing {@code
+ * originating elements are the classes or interfaces or packages (representing {@code
```
Comment (jonathan-gibbons) [inline src/java.compiler/share/classes/javax/annotation/processing/Filer.java:69]: would maybe be slightly better as "classes, interfaces or packages"

File: src/java.compiler/share/classes/javax/annotation/processing/Processor.java [hunk 2]
```diff
@@ -90,35 +90,35 @@
...
+ * <p>An annotation interface is considered present if there is at least
+ * one annotation of that interface present on an element enclosed within
  * the root elements of a round. For this purpose, a type parameter is
  * considered to be enclosed by its {@linkplain
  * TypeParameterElement#getGenericElement generic
  * element}.
 
  * For this purpose, a package element is <em>not</em> considered to
- * enclose the top-level types within that package. (A root element
+ * enclose the top-level classes and interfaces within that package. (A root element
```
Comment (jonathan-gibbons) [inline src/java.compiler/share/classes/javax/annotation/processing/Processor.java:121]: `enclose` would be better as `encloses`

**Comment by jddarcy** [comment]
First pass at updating javax.lang.model's usage of the word "type" to correspond to changes in usage of "type" being made in JLS.

The wording in annotated construct and related functionality may need another pass.

Updating @jls tags to renamed JLS sections will be done under a different bug id.

Paragraphs not reflowed initially to easy reviewing. Paragraph will be reflowed and copyright year updated before a push is done.

File: src/java.compiler/share/classes/javax/lang/model/element/TypeElement.java [hunk 3]
```diff
@@ -31,8 +31,8 @@
 
 /**
  * Represents a class or interface program element.  Provides access
- * to information about the type and its members.  Note that an enum
- * type and a record type are kinds of classes and an annotation type is a kind of
+ * to information about the class or interface and its members.  Note that an enum
+ * class and a record class are kinds of classes and an annotation interface is a kind of
```
Comment (jonathan-gibbons) [inline src/java.compiler/share/classes/javax/lang/model/element/TypeElement.java:35]: The sentence beginning "Note that..." is kinda redundant now, but not wrong.

File: src/java.compiler/share/classes/javax/lang/model/element/TypeElement.java [hunk 4]
```diff
@@ -111,23 +111,23 @@
...
      * anonymous classes, which do not have canonical names, an <a
      * href=Name.html#empty_name>empty name</a> is returned.
      *
-     * <p>The name of a generic type does not include any reference
+     * <p>The name of a generic class or interface does not include any reference
      * to its formal type parameters.
      * For example, the fully qualified name of the interface
      * {@code java.util.Set<E>} is "{@code java.util.Set}".
-     * Nested types use "{@code .}" as a separator, as in
+     * Nested class and interface use "{@code .}" as a separator, as in
```
Comment (jonathan-gibbons) [inline src/java.compiler/share/classes/javax/lang/model/element/TypeElement.java:130]: Should be "classes and interfaces"

**Review by jonathan-gibbons — CHANGES_REQUESTED** [review]
There's an instance in TypeElement.java where the plurality of the replacement is incorrect.

Other comments are mostly incidental/minor.

File: src/java.compiler/share/classes/javax/annotation/processing/Filer.java (hunk 1 above)
Comment (jddarcy) [reply src/java.compiler/share/classes/javax/annotation/processing/Filer.java:69]: Update in subsequent commit.

File: src/java.compiler/share/classes/javax/annotation/processing/Processor.java (hunk 2 above)
Comment (jddarcy) [reply src/java.compiler/share/classes/javax/annotation/processing/Processor.java:121]: "Package element" is singular so "enclose" is proper, right?

Comment (jonathan-gibbons) [reply src/java.compiler/share/classes/javax/annotation/processing/Processor.java:121]: Ooops, mea culpa; sorry

**Review by jonathan-gibbons — APPROVED** [review]

**Pull request closed without merging**